### PR TITLE
Jpeg unit

### PR DESCRIPTION
**Pull request by wudping**
implement JPEG decoderAPI unittest.

to figure out [VIZ-9126](https://jira01.devtools.intel.com/browse/VIZ-9126)

### CODE REVIEW

**Comment by wudping** [comment]
please help to review, thanks.

**Comment by xuguangxin** [comment]
Hi @uartie , please help to review.
thanks

File: decoder/vaapiDecoderJPEG.h [hunk 1]
```diff
@@ -33,6 +33,7 @@ class VaapiDecoderJPEG
     virtual YamiStatus start(VideoConfigBuffer*);
     virtual YamiStatus reset(VideoConfigBuffer*);
     virtual YamiStatus decode(VideoDecodeBuffer*);
+    YamiStatus ensureContext();
```
Comment (uartie) [inline decoder/vaapiDecoderJPEG.h]: this should be a private method

File: decoder/vaapiDecoderJPEG_unittest.cpp [hunk 2]
```diff
@@ -345,27 +345,12 @@ TEST_P(JPEGTest, Decode_SimpleMulti)
...
-
-    output = decoder.getOutput();
-    ASSERT_TRUE(bool(output));
-    EXPECT_EQ(GetParam()->getFourcc(), output->fourcc);
-
-    // Set buffer at second jpeg image
-    buffer.data = const_cast<uint8_t*>(mjpeg.data() + paramSize);
-    buffer.size = paramSize; // Length of second jpeg image data
-    buffer.timeStamp = 1;
-
```
Comment (uartie) [inline decoder/vaapiDecoderJPEG_unittest.cpp]: We should keep this check for the second image buffer decode.  The only difference now is that we don't expect to get a format change on the next decode call.  That is, notice we are testing the decode for different image in the mjpeg sequence `buffer.data = const_cast<uint8_t*>(mjpeg.data() + paramSize);`

**Review by uartie — CHANGES_REQUESTED** [review]

**Comment by uartie** [comment]
The main difference I see in this change is that now we are parsing the jpeg header twice for every image.  Is that enough to notice any performance hit?

**Comment by uartie** [comment]
previous comment edited

**Comment by uartie** [comment]
ok, maybe just the first image header is parsed twice

File: decoder/vaapiDecoderJPEG_unittest.cpp (hunk 2 above)
Comment (wudping) [reply decoder/vaapiDecoderJPEG_unittest.cpp]: start() registered for M_SOF0 is empty now, and no YAMI_DECODE_FORMAT_CHANGE signal is triggered excepted for the current resolution is really different from the previous.

Comment (wudping) [reply decoder/vaapiDecoderJPEG_unittest.cpp]: These two frames are the same.

**Review by wudping — COMMENTED** [review]
Agree with you @uartie, just the first image header is parse twice.
We don't need to call VaapiDecoderJPEG::decode() twice for every JPEG frame.

**Comment by wudping** [comment]
Updated according comment, please help to review. Thanks a lot.

File: decoder/vaapiDecoderJPEG_unittest.cpp (hunk 2 above)
Comment (uartie) [reply decoder/vaapiDecoderJPEG_unittest.cpp]: They have the same bit values, but the two frames are different data pointers.  Notice one buffer starts at "mjpeg.data()" offset and the other at "mjpeg.data() + paramSize" offset.  This verifies that the single decoder instance properly handles/decodes multiple images.  This simulates what the api user would do for mjpeg decoding.

**Review by uartie — CHANGES_REQUESTED** [review]

File: decoder/vaapiDecoderJPEG_unittest.cpp (hunk 2 above)
Comment (uartie) [reply decoder/vaapiDecoderJPEG_unittest.cpp]: That is, please decode the second image in this test and verify the expectations.

**Comment by wudping** [comment]
Thanks @uartie , the second JPEG frame in the same buffer decoding is restored to the previous.
Please help to review. Thanks again.

**Comment by wudping** [comment]
Any misunderstanding I make, please let me know, I will modify it. Thanks.

**Review by uartie — APPROVED** [review]

**Comment by xuguangxin** [comment]
thanks U.Arite

**Pull request merged**